libs/reentrancy/src/lib.sw

```diff
@@ -0,0 +1,46 @@
+//! A reentrancy check for use in Sway contracts.
+//! Note that this only works in internal contexts.
+//! to prevent reentrancy: `assert(!is_reentrant());`
+library reentrancy;
+
+use std::call_frames::*;
+use std::registers::frame_ptr;
+
+pub enum ReentrancyError {
+    NonReentrant: (),
+}
+/// Reverts if the reentrancy pattern is detected in the contract in which this is called.
+/// Not needed if the Checks-Effects-Interactions (CEI) pattern is followed (as prompted by the
+/// compiler).
+/// > Caution: While this can protect against both single-function reentrancy and cross-function
```

Comment:
I feel like this warning should be included in the README or SPECIFICATION

Reply:
Good catch, added to Known Issues.